### PR TITLE
Align constraints with upstream

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dhirschfeld** [comment]
@conda-forge-admin, please rerender

**Comment by dhirschfeld** [comment]
The main issue I was facing was with the `pandas<2.0a` constraint but I guess it makes sense so align all the deps with upstream?

https://github.com/jfcrenshaw/pzflow/blame/v3.1.1/pyproject.toml#L11-L18

**Comment by dhirschfeld** [comment]
***xref:*** https://python-poetry.org/docs/dependency-specification/#caret-requirements

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,11 +22,11 @@ requirements:
...
-    - jax
-    - optax =0.1.4
-    - pandas >=1.1,<2a0
-    - dill
-    - tqdm
+    - jax >=0.4.2,<0.5
+    - optax >=0.1.4,<0.2
+    - pandas >=1.1
+    - dill >=0.3.6,<0.4
+    - tqdm >=4.64.1,<5
```
Comment (beckermr) [inline recipe/meta.yaml:29]: Really? This code fails with tqdm greater than 5? I find that hard to believe.

Comment (beckermr) [reply recipe/meta.yaml:29]: Ditto for the other constraints.

Comment (dhirschfeld) [reply recipe/meta.yaml:29]: I don't know myself if the code will or won't work with other versions, these are just the constraints as specified by the author (who is in a better position to judge).

If the constraints are overly tight they should probably be fixed upstream by putting in a PR there and letting the upstream CI check if it breaks or not.

Comment (dhirschfeld) [reply recipe/meta.yaml:29]: I'm pretty sure all the upper bounds are greater than the latest published versions so aren't (yet) affecting anything.

Version 5, when it is published, will be a major version bump which according to semver *will* have backwards incompatible changes. Whether or not those changes will affect `pzflow` is unknowable, but it's certainly much more likely to do so than than a minor version bump.

Some people are against defensive upper bound pins but I personally am in favour as they prevent broken packages from being published, and the cost, in the `conda-forge` ecosystem at least, is simply requiring a PR to bump the pin when the new version has been published (and tested to not break the package).

**Comment by beckermr** [comment]
Well count me skeptical but I am happy to merge.

**Pull request merged**